lib/mongodb-filter-wrapper.js

```diff
@@ -0,0 +1,164 @@
+'use strict';
+
+class MongoDBFilterWrapper {
+	static get filtersAllowed() {
+		return {
+			equal: '$eq',
+			not: '$ne',
+			greater: '$gt',
+			greaterOrEqual: '$gte',
+			lesser: '$lt',
+			lesserOrEqual: '$lte',
+			in: '$in',
+			notIn: '$nin',
+			reg: '$regex',
+			all: '$all'
+		};
+	}
+
+	/**
+		* Enter door to filter process
+		* @param model
+		* @param filtersDefined
+		* @returns {{$or}|{}}
+		*/
+	static process(model, filtersDefined) {
+		let modelFilters = {};
+
+		// Get model filters
+		if(model.constructor && model.constructor.fields)
+			modelFilters = model.constructor.fields;
+
+		return MongoDBFilterWrapper.parseFiltersDefined(filtersDefined, modelFilters);
+	}
+
+	/**
+		* Validates filters given and loop calling to parse
+		* @param filtersGiven
+		* @param modelFilters
+		* @returns {{$or: Array}}
+		*/
+	static parseFiltersDefined(filtersGiven, modelFilters) {
+		let filters = {};
+		const filtersArray = [];
+		let filtersToLoop = {};
+		let orCondition = false;
+
+		if(Array.isArray(filtersGiven))
+			orCondition = true;
+
+		// Get only with objects as a filter
+		if(filtersGiven !== undefined && typeof filtersGiven === 'object' && Object.keys(filtersGiven).length > 0 && !Array.isArray(filtersGiven))
+			filtersToLoop = [filtersGiven];
+		else if(filtersGiven !== undefined && Array.isArray(filtersGiven))
+			filtersToLoop = filtersGiven;
+
+		if(Array.isArray(filtersToLoop) && filtersToLoop.length > 0) {
+			// First loop over all filters given by parameters
+			filtersToLoop.forEach(filterGiven => {
+				const formedFilter = MongoDBFilterWrapper.parseObjectFilter(filterGiven, modelFilters);
+
+				if(orCondition)
+					filtersArray.push(formedFilter);
+				else
+					filters = Object.assign(filters, formedFilter);
+
+			});
+		}
+
+		if(orCondition) {
+			return {
+				$or: filtersArray
+			};
+		}
+
+		return filters;
+	}
+
+	/**
+		* Parse the filter comparing given fields name and model fields
+		* @param filterComplete
+		* @param modelFilters
+		*/
+	static parseObjectFilter(filterComplete, modelFilters) {
+		const filterToResponse = {};
+
+		// for(const col in filterComplete) {
+		for(const [col, value] of Object.entries(filterComplete)) {
+			const field = MongoDBFilterWrapper.getFieldToCompare(col, modelFilters);
+
+			if(typeof value === 'object')
```

Comment:
Haciendo lo siguiente, no hace falta casi repetir la misma linea del `createFilter()`
```suggestion
if(typeof value !== 'object')
	value = { value };
```